contracts/pallet-ibc/src/light_clients.rs

```diff
@@ -218,11 +243,54 @@ pub enum AnyClientState {
 	Beefy(ics11_beefy::client_state::ClientState<HostFunctionsManager>),
 	#[ibc(proto_url = "TENDERMINT_CLIENT_STATE_TYPE_URL")]
 	Tendermint(ics07_tendermint::client_state::ClientState<HostFunctionsManager>),
+	#[ibc(proto_url = "WASM_CLIENT_STATE_TYPE_URL")]
+	Wasm(ics08_wasm::client_state::ClientState<AnyClient, Self, AnyConsensusState>),
 	#[cfg(test)]
 	#[ibc(proto_url = "MOCK_CLIENT_STATE_TYPE_URL")]
 	Mock(ibc::mock::client_state::MockClientState),
 }
 
+impl AnyClientState {
+	pub fn decode_recursive<F>(mut any: Any, f: F) -> Option<Self>
+	where
+		F: Fn(&Self) -> bool,
+	{
+		loop {
+			let client_state = AnyClientState::try_from(any).ok()?;
+
+			match client_state {
+				AnyClientState::Wasm(wasm_client_state) =>
+					any = Any::decode(&*wasm_client_state.data).ok()?,
+				c =>
+					if f(&c) {
```

Comment:
What’s this f? Can you please add some docs to explain?

Reply:
Added doc